logstash-core/src/main/java/org/logstash/config/ir/graph/IfVertex.java

```diff
@@ -20,6 +21,9 @@ public BooleanExpression getBooleanExpression() {
 
     public IfVertex(SourceMetadata meta, BooleanExpression booleanExpression) {
         super(meta);
+        this.id = "if-" +
+                (meta != null ? Integer.toString(meta.hashCode()) : UUID.randomUUID().toString())
```

Comment:
@ph how is it tied to the ruby lang implementation?

One downside here is that the hashCode isn't really unique enough. Java hash codes aren't intended to prevent collisions. I should probably change that.